projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.html

```diff
@@ -0,0 +1,645 @@
+                tabindex="0"
+                class="igx-filter-tree__line"
+                [ngClass]="{
+                    'igx-filter-tree__line--and': expressionItem?.operator === 0,
+                    'igx-filter-tree__line--or': expressionItem?.operator === 1,
+                    'igx-filter-tree__line--selected': expressionItem?.selected
+                }"
+                (keydown)="invokeClick($event)"
+                (click)="onGroupClick(expressionItem)"
+            ></div>
+
+            <div class="igx-filter-tree__expression">
+                <div class="igx-filter-tree__inputs" [style.display]="(isInEditMode() && this.entities.length > 0 && this.entities[0].name) ? 'flex' : 'none'">
+                    <ng-container *ngIf="!expressionItem.parent">
+                        <igx-select #entitySelect
+                            type="box"
+                            (selectionChanging)="onEntitySelectChanging($event)"
+                            [overlaySettings]="entitySelectOverlaySettings"
+                            [ngModel]="selectedEntity"
+                            [style.display]="isInEditMode() ? 'block' : 'none'"
+                        >
+                            <label igxLabel>{{ this.resourceStrings.igx_query_builder_select_entity }}</label>
+                            <igx-select-item *ngFor="let entity of entities" [value]="entity">
+                                {{entity.name}}
+                            </igx-select-item>
+                        </igx-select>
+
+                        <igx-combo
+                            #selectedReturnFieldsCombo
+                            [itemsMaxHeight]="250"
+                            [data]="fields"
+                            [displayKey]="'field'"
+                            [valueKey]="'field'"
+                            [disabled]="!selectedEntity"
+                            [(ngModel)]="selectedReturnFields"
+                            [overlaySettings]="returnFieldSelectOverlaySettings"
+                            searchPlaceholder="{{ this.resourceStrings.igx_query_builder_search }}"
+                            [style.display]="isInEditMode() ? 'block' : 'none'"
+                        >
+                            <ng-template igxComboHeader>
+                                <div
+                                    class="igx-drop-down__item igx-drop-down__item--query-builder"
+                                    (click)="onSelectAllClicked($event)"
+                                    [ngClass]="{'igx-drop-down__item--selected': this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields.length === this.selectedReturnFields.length}"
+                                >
+                                    <igx-checkbox
+                                            [checked]="this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields.length === this.selectedReturnFields.length"
```

Comment:
Forwarding Galia's suggestion:
_To be safe, we should probably check the fields before getting the length (as we do for indeterminate):_
```suggestion
                                            [checked]="this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields?.length === this.selectedReturnFields.length"
```